diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -898,6 +910,40 @@ int DifferentialFunctionComparator::cmpBasicBlocks(
                 continue;
             }
 
+            if (Reloc.status == RelocationInfo::Stored) {
+                // If there is an inequality found and we have previously found
+                // a possibly relocated block, try to match it now.
+                Reloc.status = RelocationInfo::Matching;
+                DEBUG_WITH_TYPE(DEBUG_SIMPLL,
+                                dbgs() << getDebugIndent()
+                                       << "Try to match the relocated block\n");
+                // Reset serial counters since the instructions are not equal
+                sn_mapL.erase(&*InstL);
```

Comment:
Maybe replace by a call to `DifferentialFunctionComparator::undoLastInstCompare`?